File: java/0209_Minimum_Size_Subarray_Sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int i = 0, j = 0, sum = 0;
        int result = Integer.MAX_VALUE;
        for (;j < nums.length; j++) {
            sum += nums[j];
            while (sum >= target) {
                result = Math.min(result, j - i + 1);
                sum -= nums[i++];
            }
        }
        return result == Integer.MAX_VALUE ? 0 : result;
    }
}

File: python/0209_Minimum_Size_Subarray_Sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        i = 0
        j = 0
        current_sum = 0
        result = float("inf")

        while j < len(nums):
            current_sum += nums[j]

            while current_sum >= target:
                result = min(result, j - i + 1)
                current_sum -= nums[i]
                i += 1

            j += 1
            
        return result if result != float("inf") else 0